src/csharp/Microsoft.Spark.Worker.UnitTest/DaemonWorkerTests.cs

```diff
@@ -0,0 +1,135 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Network;
+using Razorvine.Pickle;
+using Xunit;
+
+namespace Microsoft.Spark.Worker.UnitTest
+{
+    public class DaemonWorkerTests
+    {
+        [Theory]
+        [InlineData(1)]
+        [InlineData(3)]
+        public void TestsDaemonWorkerTaskRunners(int taskRunnerNumber)
```

Comment:
For each test, we are leaking threads created inside `StartServer`, right? Maybe we should either make those threads exit correctly, or create a test by launching the `Microsoft.Spark.Worker` executable. What do you think?

Reply:
Yes, we can proceed by assigning a `CancellationTokenSource` to every new thread, keep track of them using an additional dictionary collection and handle the requested cancellations through the stack, what do you think? is it worth trying to proceed in this way?